src/IROperator.h

```diff
@@ -678,6 +678,22 @@ inline Expr max(int a, const Expr &b) {
     return Internal::Max::make(Internal::make_const(b.type(), a), b);
 }
 
+/** Returns an expression representing the greater of an expressions
+ * list, after doing any necessary type coersion using
+ * \ref Internal::match_types. Vectorizes cleanly on most platforms
+ * (with the exception of integer types on x86 without SSE4).
+ * The expressions are folded from right ie. max(.., max(.., ..)). */
+inline Expr max(std::initializer_list<Expr> lst) {
+    for (auto& e : lst)
```

Comment:
style nits:
-- prefer right-cuddle to left-cuddle (auto &e)
-- always uses braces with for loops, even for single-statement loops